Libs/DICOM/Core/ctkDICOMDatabase.h

```diff
@@ -152,17 +152,23 @@ class CTK_DICOM_CORE_EXPORT ctkDICOMDatabase : public QObject
   Q_INVOKABLE QString schemaVersionLoaded();
 
   /// \brief database accessors
-  Q_INVOKABLE QStringList patients ();
+  Q_INVOKABLE QStringList patients();
   Q_INVOKABLE QStringList studiesForPatient (const QString patientUID);
   Q_INVOKABLE QStringList seriesForStudy (const QString studyUID);
   Q_INVOKABLE QStringList instancesForSeries(const QString seriesUID);
   Q_INVOKABLE QString studyForSeries(QString seriesUID);
   Q_INVOKABLE QString patientForStudy(QString studyUID);
   Q_INVOKABLE QStringList filesForSeries (const QString seriesUID);
+
   Q_INVOKABLE QHash<QString,QString> descriptionsForFile(QString fileName);
   Q_INVOKABLE QString descriptionForSeries(const QString seriesUID);
   Q_INVOKABLE QString descriptionForStudy(const QString studyUID);
   Q_INVOKABLE QString nameForPatient(const QString patientUID);
+  Q_INVOKABLE QString displayedNameForPatient(const QString patientUID);
+  Q_INVOKABLE QString fieldForPatient(const QString field, const QString patientUID);
```

Comment:
Are these fields query the display tables or the raw DICOM tables? If display then it could make sense to reflect in the name but at least in the documentation. (it is quite bad that this class is not documented but we should add some info for at lest the new methods)

Reply:
How do we get the list of fields? It would be nice to add a helper method that can provide those (or at least log the list of available fields when fieldForX fails because it gets a non-existent field).

Reply:
These functions query any field that is defined in the schema.

Currently there is no way to get the list of fields I think. Do you happen to know by heart how to get the list of fields from a table in SQL?